hazelcast-sql/src/test/java/com/hazelcast/jet/sql/impl/SqlEndToEndTestSupport.java

```diff
@@ -123,6 +125,11 @@ protected static ArrayList<Row> collectResult(SqlResult result) {
         return actualRows;
     }
 
+    @Nonnull
+    protected ObjectAssert<DAG> assertThatDag() {
+        return Assertions.<DAG>assertThat(sqlJobInvocationObserver.dag);
+    }
```

Comment:
`jobInvocationObserver.dag` and `sqlJobInvocationObserver.dag` point to DAGs that are equal for testing purposes, so choosing either of them will not make a difference. However, I think, `SqlEndToEndTestSupport` is not the best place for `JobInvocationObserverImpl` and also in the current state, it seems redundant.

Reply:
`JobInvocationObserverImpl` has the same data as `SqlJobInvocationObserverImpl` plus some more. I prefer to use higher level mechanism (SQL observer) where possible and asserting output of the component being tested instead of input to another component.
We cannot remove simply `JobInvocationObserverImpl` as some checks can be done only on deeper level: `members` chosen for query execution.